src/service.ts

```diff
@@ -41,28 +44,44 @@ export class PisaService {
         provider: ethers.providers.BaseProvider,
         wallet: ethers.Wallet,
         delayedProvider: ethers.providers.BaseProvider,
-        config?: IApiEndpointConfig
+        db: LevelUp<encodingDown<string, any>>,
+        config?: IApiEndpointConfig, 
     ) {
+        super("PISA");
         const app = express();
 
         this.applyMiddlewares(app, config);
 
+        // choose configs
+        const configs = [Raiden, Kitsune]
+
         // start reorg detector
         this.reorgDetector = new ReorgDetector(delayedProvider, 200, new ReorgHeightListenerStore());
-        this.reorgDetector.start();
 
         // dependencies
-        const store = new MemoryAppointmentStore();
+        const appointmentConstructors = new Map<ChannelType, (obj: any) => IEthereumAppointment>()
+        configs.map(c => appointmentConstructors.set(c.channelType, c.appointment));
```

Comment:
I think `for (let c of configs) { ... }` would be better here.

Reply:
I also didn't like my initialisation. I've figured out how to use the other constructor. What do you think of this:

new Map(configs.map(c => [c.channelType, c.appointment]))

I think it's quite clear and concise

Reply:
Yep, looks good.